emcc

```diff
@@ -1340,18 +1344,48 @@ try:
 
   js_transform_tempfiles = [final]
 
-  if memory_init_file:
+  if memory_init_file or shared.Settings.MEM_INIT_METHOD == 2:
     memfile = target + '.mem'
     shared.try_delete(memfile)
     def repl(m):
       # handle chunking of the memory initializer
-      s = m.groups(0)[0]
-      if len(s) == 0 and not shared.Settings.EMTERPRETIFY: return m.group(0) # emterpreter must have a mem init file; otherwise, don't emit 0-size ones
-      open(memfile, 'wb').write(''.join(map(lambda x: chr(int(x or '0')), s.split(','))))
+      s = m.group(1)
+      if len(s) == 0 and not shared.Settings.EMTERPRETIFY: return '' # emterpreter must have a mem init file; otherwise, don't emit 0-size ones
+      membytes = [int(x or '0') for x in s.split(',')]
+      while membytes and membytes[-1] == 0:
```

Comment:
Yes, stripping trailing zeros is new. At least in this context; don't know whether the `split_initializer` code affects trailing zeros as well.